compiler/optimizer/OMROptimizationManager.cpp

```diff
@@ -49,7 +49,7 @@ OMR::OptimizationManager::OptimizationManager(TR::Optimizer *o, OptimizationFact
         _numPassesCompleted(0),
         _optData(NULL),
         _optPolicy(NULL),
-        _enabled(false),
+        _enabled(!o->comp()->isDisabled(optNum)),
```

Comment:
Here is one use case I can think of that might generate different outcomes if we change how isDisabled/isEnabled works:

Opt A runs and checks to see whether opt B is disabled or not and performs differently based on the query result. Today, it only uses comp()->isDisabled(optNum) query without checking against the regex. If we add the regex check in the new isDisabled/isEnabled query, this can change what opt A does.